src/colorizer/CanvasOverlay.ts

```diff
@@ -123,62 +134,107 @@ export default class CanvasWithOverlay extends ColorizeCanvas {
+    this.canvas.setResolution(width, height);
     this.render();
   }
 
   public getIdAtPixel(x: number, y: number): number {
     const headerHeight = this.headerSize.y;
-    return super.getIdAtPixel(x, y - headerHeight);
+    return this.canvas.getIdAtPixel(x, y - headerHeight);
   }
 
   // Getters/Setters ////////////////////////////////
 
-  updateScaleBarStyle(style: Partial<ScaleBarStyle>): void {
+  public updateScaleBarStyle(style: Partial<ScaleBarStyle>): void {
     this.scaleBarStyle = { ...this.scaleBarStyle, ...style };
   }
 
-  updateTimestampStyle(style: Partial<TimestampStyle>): void {
+  public updateTimestampStyle(style: Partial<TimestampStyle>): void {
     this.timestampStyle = { ...this.timestampStyle, ...style };
   }
 
-  updateInsetBoxStyle(style: Partial<InsetBoxStyle>): void {
+  public updateInsetBoxStyle(style: Partial<InsetBoxStyle>): void {
     this.insetBoxStyle = { ...this.insetBoxStyle, ...style };
   }
 
-  updateLegendStyle(style: Partial<LegendStyle>): void {
+  public updateLegendStyle(style: Partial<LegendStyle>): void {
     this.legendStyle = { ...this.legendStyle, ...style };
   }
 
-  updateHeaderStyle(style: Partial<HeaderStyle>): void {
+  public updateHeaderStyle(style: Partial<HeaderStyle>): void {
     this.headerStyle = { ...this.headerStyle, ...style };
   }
 
-  updateFooterStyle(style: Partial<FooterStyle>): void {
+  public updateFooterStyle(style: Partial<FooterStyle>): void {
     this.footerStyle = { ...this.footerStyle, ...style };
   }
 
-  setIsExporting(isExporting: boolean): void {
+  // TODO: Move `isExporting` flag into state
+  public setIsExporting(isExporting: boolean): void {
     this.isExporting = isExporting;
   }
 
-  setCollection(collection: Collection | null): void {
-    this.collection = collection;
+  public setZoom(zoom: number): void {
+    this.zoomMultiplier = zoom;
+    if (this.canvas instanceof ColorizeCanvas) {
+      this.canvas.setZoom(zoom);
+    }
+    this.render();
+  }
+
+  public setPan(x: number, y: number): void {
+    this.panOffset.set(x, y);
+    if (this.canvas instanceof ColorizeCanvas) {
+      this.canvas.setPan(x, y);
+    }
+    this.render();
+  }
+
+  public async setParams(params: RenderCanvasStateParams): Promise<void> {
+    this.params = params;
+    await this.canvas.setParams(params);
+    this.render(false);
   }
 
-  setDatasetKey(datasetKey: string | null): void {
-    this.datasetKey = datasetKey;
+  public async setCanvas(canvas: IRenderCanvas): Promise<void> {
+    this.canvas = canvas;
+    this.canvas.setResolution(this.canvasResolution.x, this.canvasResolution.y);
+    this.canvas.setOnFrameLoadCallback(this.onFrameLoadCallback);
+    await this.canvas.setParams(this.params);
+    await this.canvas.setFrame(this.currentFrame);
+    if (this.canvas instanceof ColorizeCanvas) {
+      this.canvas.setZoom(this.zoomMultiplier);
+      this.canvas.setPan(this.panOffset.x, this.panOffset.y);
+    }
+    this.render(false);
```

Comment:
When replacing the inner canvas, copy all params + properties to it. This function is async because the inner canvas may need to load frame data before it can render for the first time.